lib/core/test/unit/Cardano/Wallet/DBSpec.hs

```diff
@@ -1,14 +1,685 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
 module Cardano.Wallet.DBSpec
     ( spec
+    , DummyTarget
+    , DummyStateMVar (..)
+    , dbPropertyTests
+    , prop_readAfterPutBoundary
```

Comment:
oh, this is for SqliteSpec. There was an error when `put . read` if status is Restoring Quantity 100 <-> Ready

Reply:
it will be used for `check that Restoring . Quantity 100 upon write results in Ready upon read` test

Reply:
Why not include it in the `dbPropertyTests`, that should hold for the MVar too no ?

Reply:
because, it fails -

Reply:
Well, something's wrong then :smile: ! Probably in the generator itself actually. Not sure that `Restoring (Quantity 100)` should be an allowed value. I thought about that the other day and we do construct the wallet restoration state "by hand". We'd probably be better off having some clear API for manipulating this type, which enforces that the status can't be above `99`.

Reply:
`Restoring (Quantity (Percentage {getPercentage = 100})` is stored as it is, but in Sqlite (at least how we implemented it) it was represented as `Ready`. This is because of how status 100 is handled in the code (maybe we will change it, and then this property is not needed)

Reply:
ok, I will remove this property and reintroduce it later (if needed) !

Reply:
ah, customizedGen makes sense - will recover it